### PR TITLE
Remove unnecessary restrictions on the library filename - fixes #276

### LINKED ISSUE #276: configure_make lib_name alnum restrictions unnecessary
The restriction of lib_name to prevent non-alnums is unnecessarily restrictive.

The following works to build a library named libconfig++, but only after the alnum restrictions are removed.

WORKSPACE
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

all_content = """filegroup(name = "all", srcs = glob(["**"]), visibility = ["//visibility:public"])"""

http_archive(
   name = "rules_foreign_cc",
   strip_prefix = "rules_foreign_cc-master",
   url = "https://github.com/bazelbuild/rules_foreign_cc/archive/master.zip",
)

load("@rules_foreign_cc//:workspace_definitions.bzl", "rules_foreign_cc_dependencies")

rules_foreign_cc_dependencies()

http_archive(
  name = "libconfig",
  build_file_content = all_content,
  strip_prefix = "libconfig-1.7.2",
  urls = ["https://github.com/werkt/libconfig/releases/download/v1.7.2/libconfig-1.7.2.tar.gz"],
)
```

BUILD
```
load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")

configure_make(
    name = "libconfigpp",
    lib_name = "libconfig++",
    lib_source = "@libconfig//:all",
    out_lib_dir = "lib",
)
```

### PR DESCRIPTION
**Pull request by irengrig**

### CODE REVIEW

**Comment by irengrig** [comment]
Inspired by https://stackoverflow.com/questions/4814040/allowed-characters-in-filename

File: tools/build_defs/framework.bzl [hunk 1]
```diff
@@ -462,16 +462,15 @@ def _symlink_contents_to_dir(dir_name, files_list):
...
+_FORBIDDEN_FOR_FILENAME = ["\\", "/", ":", "*", "\"", "<", ">", "|"]
+
 def _check_file_name(var, name):
     if (len(var) == 0):
-        fail("{} can not be empty string.".format(name.capitalize()))
-
-    if (not var[0:1].isalpha()):
-        fail("{} should start with a letter.".format(name.capitalize()))
-    for index in range(1, len(var) - 1):
+        fail("{} can not be an empty string.".format(name.capitalize()))
```
Comment (jin) [inline tools/build_defs/framework.bzl]:
```suggestion
        fail("{} cannot be an empty string.".format(name.capitalize()))
```

File: tools/build_defs/framework.bzl [hunk 2]
```diff
@@ -462,16 +462,15 @@ def _symlink_contents_to_dir(dir_name, files_list):
...
+    for index in range(0, len(var) - 1):
         letter = var[index]
-        if not letter.isalnum() and letter != "_":
-            fail("{} should be alphanumeric or '_'.".format(name.capitalize()))
+        if letter in _FORBIDDEN_FOR_FILENAME:
+            fail("Symbol %s is forbidden in %s." % (letter, name.capitalize()))
```
Comment (jin) [inline tools/build_defs/framework.bzl]: why do we capitalize the name? also, please surround the `%s` template strings in escaped quotes so it reads like: `Symbol ':' is forbidden in the name 'foo:bar'`.

```suggestion
            fail("Symbol '%s' is forbidden in name '%s'." % (letter, name.capitalize()))
```

Comment (irengrig) [reply tools/build_defs/framework.bzl]: Please note it is not the name of the library itself, ... anyway, I inlined the "library name text".

**Pull request merged**